app/routers/users.py

```diff
@@ -33,7 +33,7 @@ async def list_user_servers(current_user: User = Depends(get_current_user)):
 @router.patch("/me", response_model=EitherUserProfile, summary="Update user profile")
 async def patch_update_user_profile(
     server_id: Optional[str] = None,
-    update_data: Union[ServerMemberUpdateSchema, UserUpdateSchema] = Body(...),
+    update_data: Union[UserUpdateSchema, ServerMemberUpdateSchema] = Body(...),
```

Comment:
In this case, we have 2 diff schemas for the `update_data` field of the request. FastAPI will try to match the request body to the first one in the list, and then the second one. So, we need to add the more explicit schema (the one w/ more fields) first and then the more generic one.

In this case, UserUpdateSchema has 3 fields: `display_name`, `pfp`, and `description`; whilst the ServerMemberUpdateSchema has 2 fields: `display_name` and `pfp`. That's why it wasn't picking up the `description` field in your tests.